### PR TITLE
Add SO8145 lambda function

### PR DESCRIPTION
**Pull request by christofile**
*Issue #, if available:*

N/A

*Description of changes:*
  (SO8145)
  cache-control-with-regex-path
  
  The lambda shows how to change cache-control of origin response using path matching with regular expression.
  
*How Has This Been Tested:*
1. Unit test
    Run "npm test" locally and the unit test is passed
3. Deployment test
    Run "sam build %% sam deploy -g --capabilities CAPABILITY_NAMED_IAM" and the deployment is succeeded.
4. Solution id
    My solution id (SO8145) is in the output

*[x] My testing has passed*

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by yanbasic** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: nodejs-build-cloudfront-extensions
* Commit ID: 2b7dd8b479cbb87143ec702622db7a157e99c165
* Result: FAILED
* [Build Logs](https://uc0zdtvc9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=75ecad4b-e72f-49ca-9193-f728ec4557a9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: edge/nodejs/cache-control-with-regex-path/template.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            Action: sts:AssumeRole
+        ManagedPolicyArns:
+          - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+
+  CacheControlRegexPathFunction:
+    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Properties:
+      CodeUri: cache-control-with-regex-path/
+      Handler: app.handler
+      Runtime: nodejs14.x
```
Comment (NingLu) [inline edge/nodejs/cache-control-with-regex-path/template.yaml:52]: Better to use the latest nodejs version - nodejs16.x

File: edge/nodejs/cache-control-with-regex-path/cache-control-with-regex-path/package.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "hello_world",
```
Comment (NingLu) [inline edge/nodejs/cache-control-with-regex-path/cache-control-with-regex-path/package.json:2]: Update it to your app's name

File: edge/nodejs/cache-control-with-regex-path/cache-control-with-regex-path/package.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "version": "1.0.0",
+  "description": "hello world sample for NodeJS",
```
Comment (NingLu) [inline edge/nodejs/cache-control-with-regex-path/cache-control-with-regex-path/package.json:4]: Update the description

File: edge/nodejs/cache-control-with-regex-path/cache-control-with-regex-path/package.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  "main": "app.js",
+  "repository": "https://github.com/awslabs/aws-sam-cli/tree/develop/samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs",
```
Comment (NingLu) [inline edge/nodejs/cache-control-with-regex-path/cache-control-with-regex-path/package.json:6]: Update the url to CF Extensions github url, such as 

https://github.com/awslabs/aws-cloudfront-extensions/tree/main/edge/nodejs/cache-control-with-regex-path

**Comment by christofile** [comment]
@NingLu I've updated the code.

**Comment by yanbasic** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: nodejs-build-cloudfront-extensions
* Commit ID: 89e57a2c719517d03f493d21c5ccd96d4f7e3df5
* Result: FAILED
* [Build Logs](https://uc0zdtvc9j.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=968dfa85-0b48-4a8a-9fbd-8138db071264%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**